pdp/policy.go

```diff
@@ -27,8 +30,12 @@ var (
 	// of the algorithm. Contains only algorithms which require parameters.
 	RuleCombiningParamAlgs = map[string]RuleCombiningAlgMaker{
 		"mapper": makeMapperRCA}
+
+	errHiddenPolicy = fmt.Errorf("Attempting to marshal hidden policy")
```

Comment:
While we have only several marshaling functions an error can appear at any nesting level. It is good to log where exactly it goes from. PDP has specific approach implemented with boundError struct (see error.go file). To use it you need to define your specific error with error.yaml (there are a lot of examples here so I hope you can get in easily). `go generate` to update errors.go file after changes in errors.yaml. Then you need to make error with generated new&lt;WhateverYouPutHere&gt; function and instead of `return err` put `return bindError(err, "whatever context")` or `return bindErrorf(err, "what%s context", "ever")`.